mlir/include/mlir/Dialect/Linalg/IR/LinalgEnums.td

```diff
@@ -15,6 +15,57 @@
 
 include "mlir/IR/EnumAttr.td"
 
+// Define an `enum class : i32` to categorise element-wise op.
+def ElemwiseNAryCategory : I32EnumAttr<"ElemwiseNAryCategory", "", [
```

Comment:
This might be frowned upon, but I dont think we need two different sets of enums. We could just use a range of enums in `ElemwiseFn` below to denote unary/binary and ternary functions....